CIDER/cider.py

```diff
@@ -230,6 +230,55 @@ def import_as_data_dict(self, path_to_data: str) -> Dict:
                 )
         return all_dicts
 
+    def import_smi_as_data_dict(self, path_to_data: str):
+        """
+        This function creates a dictionary with the names of the imported file as keys. The values of each of these keys is a subdictionary. The first entry of every subdictionary is self.import_keyname (class variable, can be changed) as key and rdkit_mol Objects (list of rdkit.Chem.rdchem.Mol objects) of the SMI File as value. To find faulty molecules, every entry of the rdkit_mol Objects will be parsed once. (Parsed molecules will not be stored in the dictionary to save memory.)
+
+        Args:
+            path_to_data (str): Path to the directory where the SMI Files are stored.
+
+        Returns:
+            all_dicts (dict): Dictionary with subdictionaries for every dataset, including the key 'self.import_keyname'.
+
+        Raises:
+            FileNotFoundError: if the data path is invalid.
+            KeyError: if no SMI Files are in the given directory.
+        """
+        all_dicts = {}
+        data_dir = os.path.abspath(str(path_to_data))
+        for dict_name in os.listdir(data_dir):
+            if dict_name[-3:] == "smi" or dict_name[-3:] == "SMI":
```

Comment:
Sometimes, SMILES files are also simply stored as .txt. I would therefore include that here. What do the others think?

Reply:
I would agree with you. Most people also store SMILES without any extension as well.

Reply:
How can you store a file without *any* extension? It would be a text file then, right?